### PR TITLE
ci: remove read access from kube config

### LINKED ISSUE #8169: ci: helm test are failing frequently
**Is this a bug report or feature request?**
* Bug Report

```Errors Encountered while executing helm command . WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/runner/.kube/config```. This error is coming from the helm only.

**Deviation from expected behavior:**
Integration test CephHelmSuite actions are failing
**Expected behavior:**
Integration test CephHelmSuite actions should pass

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit remove read access from kube
config file for group and others.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #8169 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
passing 2/2

**Comment by subhamkrai** [comment]
seeing new error `coalesce.go:199: warning: destination for config is a table. Ignoring non-table value <nil>` looking

**Comment by subhamkrai** [comment]
with the current changes, we are not getting errors like this which fails the helm suits
earlier,
```
make -C images/ceph csv-clean
=== ensuring modules are tidied
=== installing helm
=== helm package rook-ceph
=== helm package rook-ceph-cluster
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/runner/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/runner/.kube/config
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/runner/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/runner/.kube/config
==> Linting /home/runner/work/rook/rook/_output/rook-ceph-cluster
```
Now,
```
make -j IMAGES=ceph VERSION=0 build
make -C images/ceph csv-clean
=== installing helm
=== ensuring modules are tidied
=== helm package rook-ceph
=== helm package rook-ceph-cluster
==> Linting /home/runner/work/rook/rook/_output/rook-ceph-cluster
```

**Comment by subhamkrai** [comment]
For now, we are good but in the future, we'll face few errors with the new version of helm and minikube.
Errors we could face in future
1. `coalesce.go:199: warning: destination for config is a table. Ignoring non-table value <nil>` which basically mean we can't keep the fields empty [here](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph-cluster/values.yaml#L218) simillary with other fields too like network, annotations....
2. 
```
Run tests/scripts/minikube.sh helm
+ tests/scripts/minikube.sh helm
 copying rook image for helm
X Exiting due to ENV_DRIVER_CONFLICT: 'none' driver does not support 'minikube docker-env' command
Error: Process completed with exit code 1.
```
we face this with the latest minikube version

**Comment by leseb** [comment]
> For now, we are good but in the future, we'll face few errors with the new version of helm and minikube.
> Errors we could face in future
> 
> 1. `coalesce.go:199: warning: destination for config is a table. Ignoring non-table value <nil>` which basically mean we can't keep the fields empty [here](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph-cluster/values.yaml#L218) simillary with other fields too like network, annotations....

So we need to comment it out?

> 
> ```
> Run tests/scripts/minikube.sh helm
> + tests/scripts/minikube.sh helm
>  copying rook image for helm
> X Exiting due to ENV_DRIVER_CONFLICT: 'none' driver does not support 'minikube docker-env' command
> Error: Process completed with exit code 1.
> ```
> 
> we face this with the latest minikube version

This won't affect the CI.

File: .github/workflows/integration-test-helm-suite.yaml [hunk 1]
```diff
@@ -35,6 +35,9 @@ jobs:
         start args: --memory 6g --cpus=2
         github token: ${{ secrets.GITHUB_TOKEN }}
 
+    - name: update permission
```
Comment (leseb) [inline .github/workflows/integration-test-helm-suite.yaml:38]:
```suggestion
    - name: remove read permission from kube config file

File: .github/workflows/integration-tests-on-release.yaml [hunk 2]
```diff
@@ -87,6 +87,9 @@ jobs:
         start args: --memory 6g --cpus=2
         github token: ${{ secrets.GITHUB_TOKEN }}
 
+    - name: update permission
```
Comment (leseb) [inline .github/workflows/integration-tests-on-release.yaml:90]: ditto

**Review by leseb — CHANGES_REQUESTED** [review]

**Comment by subhamkrai** [comment]
> > For now, we are good but in the future, we'll face few errors with the new version of helm and minikube.
> > Errors we could face in future
> > 
> > 1. `coalesce.go:199: warning: destination for config is a table. Ignoring non-table value <nil>` which basically mean we can't keep the fields empty [here](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph-cluster/values.yaml#L218) simillary with other fields too like network, annotations....
> 
> So we need to comment it out?
or we can pass do `{}` this

> 
> > ```
> > Run tests/scripts/minikube.sh helm
> > + tests/scripts/minikube.sh helm
> >  copying rook image for helm
> > X Exiting due to ENV_DRIVER_CONFLICT: 'none' driver does not support 'minikube docker-env' command
> > Error: Process completed with exit code 1.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > we face this with the latest minikube version
> 
> This won't affect the CI.

It is affecting and I was wondering why because we have this since start but with latest version it is failing https://github.com/rook/rook/runs/2911768042?check_suite_focus=true see issue #8202

**Comment by leseb** [comment]
> > > For now, we are good but in the future, we'll face few errors with the new version of helm and minikube.
> > > Errors we could face in future
> > > 
> > > 1. `coalesce.go:199: warning: destination for config is a table. Ignoring non-table value <nil>` which basically mean we can't keep the fields empty [here](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph-cluster/values.yaml#L218) simillary with other fields too like network, annotations....
> > 
> > 
> > So we need to comment it out?
> > or we can pass do `{}` this
> 
> > > ```
> > > Run tests/scripts/minikube.sh helm
> > > + tests/scripts/minikube.sh helm
> > >  copying rook image for helm
> > > X Exiting due to ENV_DRIVER_CONFLICT: 'none' driver does not support 'minikube docker-env' command
> > > Error: Process completed with exit code 1.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > we face this with the latest minikube version
> > 
> > 
> > This won't affect the CI.
> 
> It is affecting and I was wondering why because we have this since start but with latest version it is failing https://github.com/rook/rook/runs/2911768042?check_suite_focus=true see issue #8202

The error makes sense, if we use the `none` driver using `minikube docker-env` makes no sense. So we should just remove it from our script.

File: .github/workflows/integration-test-helm-suite.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/integration-test-helm-suite.yaml]: done.

**Comment by subhamkrai** [comment]
> > > > For now, we are good but in the future, we'll face few errors with the new version of helm and minikube.
> > > > Errors we could face in future
> > > > 
> > > > 1. `coalesce.go:199: warning: destination for config is a table. Ignoring non-table value <nil>` which basically mean we can't keep the fields empty [here](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph-cluster/values.yaml#L218) simillary with other fields too like network, annotations....
> > > 
> > > 
> > > So we need to comment it out?
> > > or we can pass do `{}` this
> > 
> > 
> > > > ```
> > > > Run tests/scripts/minikube.sh helm
> > > > + tests/scripts/minikube.sh helm
> > > >  copying rook image for helm
> > > > X Exiting due to ENV_DRIVER_CONFLICT: 'none' driver does not support 'minikube docker-env' command
> > > > Error: Process completed with exit code 1.
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > we face this with the latest minikube version
> > > 
> > > 
> > > This won't affect the CI.
> > 
> > 
> > It is affecting and I was wondering why because we have this since start but with latest version it is failing https://github.com/rook/rook/runs/2911768042?check_suite_focus=true see issue #8202
> 
> The error makes sense, if we use the `none` driver using `minikube docker-env` makes no sense. So we should just remove it from our script.

okay, I'll change in follow up PR where I'll be updating minikube version and also create PR for updating helm version

**Review by leseb — APPROVED** [review]

File: tests/framework/utils/helm_helper.go [hunk 3]
```diff
@@ -18,11 +18,12 @@ package utils
 
 import (
 	"fmt"
-	"gopkg.in/yaml.v2"
```
Comment (travisn) [inline tests/framework/utils/helm_helper.go:21]: Could this file be reverted? It looks like nothing really changed and not related to the PR?

Comment (subhamkrai) [reply tests/framework/utils/helm_helper.go]: done

**Pull request merged**